src/bidiMapper/domains/script/scriptEvaluator.ts

```diff
@@ -315,6 +363,48 @@ export class ScriptEvaluator {
     argumentValue: Script.ArgumentValue,
     realm: Realm
   ): Promise<Protocol.Runtime.CallArgument> {
+    if ('sharedId' in argumentValue) {
+      const [navigableId, rawBackendNodeId] = argumentValue.sharedId.split(
+        ScriptEvaluator.#SHARED_ID_DIVIDER
+      );
+
+      const backendNodeId = parseInt(rawBackendNodeId ?? '');
+      if (
+        isNaN(backendNodeId) ||
+        backendNodeId === undefined ||
+        navigableId === undefined
+      ) {
+        throw new Message.InvalidArgumentException(
+          `SharedId "${
+            argumentValue.sharedId
+          }" should have format "{navigableId}${
+            ScriptEvaluator.#SHARED_ID_DIVIDER
+          }{backendNodeId}".`
+        );
+      }
+
+      if (realm.navigableId !== navigableId) {
+        throw new Message.NoSuchNodeException(
+          `SharedId "${argumentValue.sharedId}" belongs to different document.`
+        );
+      }
+
+      try {
+        const obj = await realm.cdpClient.sendCommand('DOM.resolveNode', {
+          backendNodeId,
+          executionContextId: realm.executionContextId,
+        });
+        // TODO: add `obj.object.objectId` to GC.
+        return {objectId: obj.object.objectId};
+      } catch (e: any) {
```

Comment:
In general, use `unknown` in `try/catch` blocks, instead of `any`

https://google.github.io/styleguide/tsguide.html#exceptions

Reply:
you cannot access some random fields of the `unknown`